src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -42,80 +47,138 @@ var kubeconfigCmd = &cobra.Command{
+		linuxConfigDir := path.Join(homedir.HomeDir(), ".kube")
+		linuxConfig, err := readKubeConfig(filepath.Join(linuxConfigDir, "config"))
+		if err != nil {
+			return err
+		}
+
+		cleanConfig := removeExistingRDConfig(rdCluster, &linuxConfig)
+
+		kubeConfig, err := updateClusterIP(winConfig, *cleanConfig, rdNetworking)
+
+		var finalKubeConfigFile *os.File
+		if err := os.MkdirAll(linuxConfigDir, 0o750); err != nil {
+			return err
+		}
+		finalKubeConfigFile, err = os.Create(filepath.Join(linuxConfigDir, "config"))
+		if err != nil {
+			return err
+		}
+		defer finalKubeConfigFile.Close()
+		err = os.MkdirAll(linuxConfigDir, 0o750)
+		if err != nil && !errors.Is(err, os.ErrExist) {
+			// The error already contains the full path, we can't do better.
+			return err
+		}
+		err = yaml.NewEncoder(finalKubeConfigFile).Encode(kubeConfig)
+		if err != nil {
+			return err
 		}
 		return nil
 	},
 }
 
+func readKubeConfig(configPath string) (kubeConfig, error) {
+	var config kubeConfig
+	configFile, err := os.Open(configPath)
+	if err != nil {
+		return config, err
+	}
+	defer configFile.Close()
+	err = yaml.NewDecoder(configFile).Decode(&config)
+	if err != nil {
+		return config, err
+	}
+
+	return config, nil
+}
+
+func updateClusterIP(winConfig, linuxConfig kubeConfig, rdNetworking bool) (kubeConfig, error) {
+	ip, err := getClusterIP()
+	if err != nil {
+		return winConfig, err
+	}
+	// Fix up any clusters at 127.0.0.1, using the IP address we found.
+	for clusterIdx, cluster := range winConfig.Clusters {
+		server, err := url.Parse(cluster.Cluster.Server)
+		if err != nil {
+			// Ignore any clusters with invalid servers
+			continue
+		}
+		if winConfig.Clusters[clusterIdx].Name != rdCluster {
+			continue
+		}
+		if rdNetworking {
+			server.Host = net.JoinHostPort("gateway.rancher-desktop.internal", server.Port())
+		} else {
+			if server.Port() != "" {
+				server.Host = net.JoinHostPort(ip.String(), server.Port())
+			} else {
+				server.Host = ip.String()
+			}
+
+		}
+		winConfig.Clusters[clusterIdx].Name = rdCluster
+		winConfig.Clusters[clusterIdx].Cluster.Server = server.String()
+	}
+	return mergeKubeConfigs(winConfig, linuxConfig), nil
+}
+
+func removeExistingRDConfig(clusterName string, config *kubeConfig) *kubeConfig {
+	var newClusters []struct {
+		Cluster struct {
+			Server string
+			Extras map[string]interface{} `yaml:",inline"`
+		} `yaml:"cluster"`
+		Name   string                 `yaml:"name"`
+		Extras map[string]interface{} `yaml:",inline"`
+	}
+
+	for _, cluster := range config.Clusters {
+		if cluster.Name != clusterName {
+			newClusters = append(newClusters, cluster)
+		}
+	}
+
+	config.Clusters = newClusters
+	return config
+}
+
+func mergeKubeConfigs(winConfig, linuxConfig kubeConfig) kubeConfig {
+	winConfig.Clusters = append(winConfig.Clusters, linuxConfig.Clusters...)
+	winConfig.Contexts = append(winConfig.Contexts, linuxConfig.Contexts...)
+	winConfig.Users = append(winConfig.Users, linuxConfig.Users...)
```

Comment:
We don't want _all_ of the Windows configs, just the `rancher-desktop` ones, right?